### ISSUE TITLE
[no-unnecessary-condition] False positive for nested arrays

### ISSUE THREAD

**Issue by SCG82** [typescript-eslint/typescript-eslint#2255]
**Repro**

```JSON
{
 "extends": [
    "plugin:@typescript-eslint/eslint-recommended",
    "plugin:@typescript-eslint/recommended",
    "plugin:@typescript-eslint/recommended-requiring-type-checking"
  ],
  "rules": {
    "@typescript-eslint/no-unnecessary-condition": "warn"
  }
}
```

```TS
const latencies: number[][] = [];

function recordData(): void {
    if (!latencies[0])
        latencies[0] = [];
    latencies[0].push(4);
}

recordData();
```

**Expected Result**
No errors or warnings.
**Actual Result**
```
4:7  warning  Unnecessary conditional, value is always falsy  @typescript-eslint/no-unnecessary-condition
```
**Additional Info**

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `3.4.0` |
| `@typescript-eslint/parser`        | `3.2.0` |
| `TypeScript`                       | `3.9.5` |
| `ESLint`                           | `7.2.0` |
| `node`                             | `12.18.1` |
| `npm`                              | `6.14.5` |

**Comment by renugasaraswathy** [comment]
I will take up this issue.
It is not just for nested arrays, it is because the type of unaryExpression is boolean. I will add a handler for UnaryExpression here and raise a PR.
https://github.com/typescript-eslint/typescript-eslint/blob/adcee02d779f1faade877cb4a2692af5141ef09f/packages/eslint-plugin/src/rules/no-unnecessary-condition.ts#L567-L577

**Comment by renugasaraswathy** [comment]
```js
const a=2;
if(a) { // This behaves as expected
}
```
```js
const a=2;
if(-a) { // This is always truthy, but not caught by the rule
}
```
```js
const a=2;
if(+a) { // This is always truthy, but not caught by the rule
}
```

In general, when an unary expression is present, there are false positives, I think it has to be handled separately. 
I have given PR for unary negation expression.
